src/libraries/TypesLib.sol

```diff
@@ -1,40 +1,42 @@
 // SPDX-License-Identifier: MIT
-pragma solidity 0.8.24;
+pragma solidity ^0.8;
 
 import "./BLS.sol";
 
+/// @title TypesLib
+/// @author Randamu
+/// @notice Library declaring custom data types used for randomness and blocklock requests
 library TypesLib {
-    // Signature request struct for signature request type
-    struct SignatureRequest {
-        bytes message; // plaintext message to hash and sign
-        bytes messageHash; // hashed message to sign
-        bytes condition; // optional condition, length can be zero for immediate message signing
-        string schemeID; // signature scheme id, e.g., "BN254", "BLS12-381", "TESS"
-        address callback; // the requester address to call back. Must implement ISignatureReceiver interface to support the required callback
+    /// @notice  Ciphertext representing data encrypted off-chain
+    struct Ciphertext {
+        BLS.PointG2 u;
+        bytes v;
+        bytes w;
     }
 
-    // Blocklock request stores details needed to generate blocklock decryption keys
+    /// @notice  Blocklock request stores details needed to generate blocklock decryption keys
     struct BlocklockRequest {
-        uint256 decryptionRequestID;
+        uint256 subId; // must be 0 for direct funding
+        uint256 directFundingFeePaid; // must be > 0 for direct funding and if subId == 0
+        uint64 decryptionRequestID;
         uint256 blockHeight;
         Ciphertext ciphertext;
         bytes signature;
+        bytes decryptionKey;
         address callback;
     }
 
-    struct Ciphertext {
-        BLS.PointG2 u;
-        bytes v;
-        bytes w;
-    }
-
-    // Decryption request stores details for each decryption request
+    /// @notice  Decryption request stores details for each decryption request
     struct DecryptionRequest {
         string schemeID; // signature scheme id, e.g., "BN254", "BLS12-381", "TESS"
```

Comment:
perhaps our examples here ought to be real schemes just to as not to confuse any noob readers

Reply:
Change to only BN254 with a brief description of the scheme in `ISignatureScheme.sol`